File: uri/1036.cpp
#include <iostream>
#include <cmath>

int main(){
    double A,B,C,delta,r1,r2;
    
    std::cin>>A>>B>>C;

    if(A==0){std::cout<<"Impossivel calcular"<<std::endl;return 0;}

    delta = (B*B)-(4*A*C);

    if(delta<0){std::cout<<"Impossivel calcular"<<std::endl;return 0;}

    r1 = (-B+sqrt(delta))/(2*A);
    r2 = (-B-sqrt(delta))/(2*A);

    std::cout.precision(5);
    std::cout<<std::fixed;

    std::cout<<"R1 = "<<r1<<std::endl;
    std::cout<<"R2 = "<<r2<<std::endl;
}

File: uri/rust/1036.rs
fn get_float_vec() -> Vec<f64> {
    let mut buffer = String::new();
    std::io::stdin().read_line(&mut buffer).expect("Failed");
    let split = buffer.split_whitespace();
    let vec: Vec<&str> = split.collect();
    vec.iter().map(|x| x.trim().parse::<f64>().unwrap()).collect()
}

fn main() {
    let v: Vec<f64> = get_float_vec();
    let dt: f64 = (v[1].powf(2.0))-(4.0*v[0]*v[2]);

    if v[0] == 0.0 || dt < 0.0 {
        println!("Impossivel calcular");
    } else {
        println!("R1 = {:.5}", (-v[1]+dt.sqrt())/(2.0*v[0]) );
        println!("R2 = {:.5}", (-v[1]-dt.sqrt())/(2.0*v[0]) );
    }
}
